src/main/java/uk/gov/hmcts/cmc/claimstore/services/pilotcourt/PilotCourtService.java

```diff
@@ -0,0 +1,115 @@
+import org.apache.commons.csv.CSVFormat;
+import org.apache.commons.csv.CSVParser;
+import org.apache.commons.csv.CSVRecord;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.cmc.claimstore.courtfinder.CourtFinderApi;
+import uk.gov.hmcts.cmc.claimstore.services.ccd.legaladvisor.HearingCourt;
+import uk.gov.hmcts.cmc.claimstore.services.ccd.legaladvisor.HearingCourtMapper;
+import uk.gov.hmcts.cmc.claimstore.utils.ResourceReader;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import javax.annotation.PostConstruct;
+
+@Service
+public class PilotCourtService {
+
+    private final Logger logger = LoggerFactory.getLogger(this.getClass());
+
+    private Map<String, PilotCourt> pilotCourts;
+    private final CourtFinderApi courtFinderApi;
+    private final HearingCourtMapper hearingCourtMapper;
+    private final String dataSource;
+
+    public PilotCourtService(@Value("${pilot-courts.datafile}") String dataSource,
+                             CourtFinderApi courtFinderApi,
+                             HearingCourtMapper hearingCourtMapper) {
+        this.dataSource = dataSource;
+        this.hearingCourtMapper = hearingCourtMapper;
+        this.courtFinderApi = courtFinderApi;
+    }
+
+    @PostConstruct
+    public void init() {
+
+        try {
+            String data = ResourceReader.readString(dataSource);
+            CSVParser parse = CSVParser.parse(data, CSVFormat.DEFAULT);
+            Iterator<CSVRecord> iterator = parse.iterator();
+
+            Map<String, PilotCourt> pilotCourts = buildPilotCourts(iterator);
+            this.pilotCourts = Collections.unmodifiableMap(pilotCourts);
+
+        } catch (IOException e) {
+            throw new IllegalStateException(e);
+        }
+    }
+
+    public Set<String> getAllPilotCourtIds() {
+        return pilotCourts.keySet();
+    }
+
+    public HearingCourt getHearingCourt(String pilotCourtId) {
+        if (!pilotCourts.containsKey(pilotCourtId)) {
+            throw new IllegalArgumentException(String.format("Supplied pilotCourtId '%s' not found", pilotCourtId));
+        }
+
+        PilotCourt pilotCourt = pilotCourts.get(pilotCourtId);
+
+        return pilotCourt.getHearingCourt().orElseGet(() -> {
+                Optional<HearingCourt> court = getCourt(pilotCourt.getPostcode());
+                pilotCourt.setHearingCourt(court);
+
+                return court.orElseGet(() -> HearingCourt.builder().build());
+            }
+        );
+    }
+
+    public boolean isPilotCourt(String courtName) {
+        if (courtName == null) {
+            return false;
+        }
+
+        return pilotCourts.keySet().stream().anyMatch(pilotCourt -> courtName.toUpperCase().contains(pilotCourt));
```

Comment:
There's a `StringUtils.containsIgnoreCase()` which might be suitable, from apache commons; but is `contains` really appropriate here?

Reply:
Agree on the containsIngoreCase(). This is how we currently allow people to chose a different court via the free text input on the frontend. I'm not too happy about this and would like to go over this functionality with Marc as it leaves us open to not catching pilot claims.

Reply:
Also this code is changing in a further story. The PR is almost ready.